pallets/asset/src/checkpoint/migrations.rs

```diff
@@ -0,0 +1,144 @@
+                double_map hasher(blake2_128_concat) Ticker, hasher(twox_64_concat) CheckpointId => polymesh_primitives::Balance;
+
+            // This storage changed the Ticker key to AssetID.
+            pub Balance get(fn balance_at_checkpoint):
+                double_map hasher(blake2_128_concat) (Ticker, CheckpointId), hasher(twox_64_concat) IdentityId => polymesh_primitives::Balance;
+
+            // This storage changed the Ticker key to AssetID.
+            pub CheckpointIdSequence get(fn checkpoint_id_sequence):
+                map hasher(blake2_128_concat) Ticker => CheckpointId;
+
+            // This storage changed the Ticker key to AssetID.
+            pub BalanceUpdates get(fn balance_updates):
+                double_map hasher(blake2_128_concat) Ticker, hasher(twox_64_concat) IdentityId => Vec<CheckpointId>;
+
+            // This storage changed the Ticker key to AssetID.
+            pub Timestamps get(fn timestamps):
+                double_map hasher(blake2_128_concat) Ticker, hasher(twox_64_concat) CheckpointId => Moment;
+
+            // This storage changed the Ticker key to AssetID.
+            pub ScheduleIdSequence get(fn schedule_id_sequence):
+                map hasher(blake2_128_concat) Ticker => ScheduleId;
+
+            // This storage changed the Ticker key to AssetID.
+            pub CachedNextCheckpoints get(fn cached_next_checkpoints):
+                map hasher(blake2_128_concat) Ticker => Option<NextCheckpoints>;
+
+            // This storage changed the Ticker key to AssetID.
+            pub ScheduledCheckpoints get(fn scheduled_checkpoints):
+                double_map hasher(blake2_128_concat) Ticker, hasher(twox_64_concat) ScheduleId => Option<ScheduleCheckpoints>;
+
+            // This storage changed the Ticker key to AssetID.
+            pub ScheduleRefCount get(fn schedule_ref_count):
+                double_map hasher(blake2_128_concat) Ticker, hasher(twox_64_concat) ScheduleId => u32;
+
+            // This storage changed the Ticker key to AssetID.
+            pub SchedulePoints get(fn schedule_points):
+                double_map hasher(blake2_128_concat) Ticker, hasher(twox_64_concat) ScheduleId => Vec<CheckpointId>;
+        }
+    }
+
+    decl_module! {
+        pub struct Module<T: Config> for enum Call where origin: T::RuntimeOrigin { }
+    }
+}
+
+pub(crate) fn migrate_to_v2<T: Config>() {
+    RuntimeLogger::init();
+    let mut ticker_to_asset_id = BTreeMap::new();
+
+    // Removes all elements in the old storage and inserts it in the new storage
+
+    log::info!("Updating types for the TotalSupply storage");
+    v1::TotalSupply::drain().for_each(|(ticker, checkpoint_id, balance)| {
+        let asset_id = ticker_to_asset_id
+            .entry(ticker)
+            .or_insert(AssetID::from(ticker));
+        TotalSupply::insert(asset_id, checkpoint_id, balance);
+    });
+
+    log::info!("Updating types for the Balance storage");
```

Comment:
I recommend counting how many items are migrated and printing the count.  The nodes will log this information and we can use it to double check that the correct number of items have been migrated.